index.js

```diff
@@ -74,15 +74,25 @@ module.exports = function (processors, options) {
     }
 
     function handleError (error) {
-      var errorOptions = { fileName: file.path, showStack: true }
-      if (error.name === 'CssSyntaxError') {
-        error = error.message + '\n\n' + error.showSourceCode() + '\n'
-        errorOptions.showStack = false
+      var errorOptions = { fileName: file.path }
+      if (error instanceof Object) {
+        errorOptions.error = error
+        if (error.name === 'CssSyntaxError') {
+          var input = error.input || {};
+          errorOptions.fileName = error.file || input.file || file.path
+          errorOptions.lineNumber = error.line || input.line
+          errorOptions.showStack = false
+          errorOptions.showProperties = false
+          error = error.message + '\n\n' + error.showSourceCode() + '\n'
+        } else {
+          errorOptions.showStack = true
+          error = error.message || error
+        }
       }
       // Prevent stream’s unhandled exception from
       // being suppressed by Promise
       setImmediate(function () {
-        cb(new gutil.PluginError('gulp-postcss', error, errorOptions))
+        cb(new gutil.PluginError('gulp-postcss', String(error), errorOptions))
```

Comment:
According to docs `gutil.PluginError` may take Error object in a second argument. So there is no need to convert it to a string.